---
title: "Software and Workshops"
---

### Workshop Materials

[Center for Biodiversity Analysis: Gen3sis Workshop 2024](https://alexskeels.github.io/CBAGen3sis)

![Flier for the workshop](pictures/gen3sis.jpg)

### R Tools

Dynamic Range Evolution And Diversification model ([DReAD](https://github.com/alexskeels/DynamicRangeEvolutionAndDiversification))

-   Inferring the geographic mode of speciation could help reveal the evolutionary and ecological mechanisms that underlie the generation of biodiversity. Comparative methods have sought to reconstruct the geographic speciation history of clades using data on phylogeny and species geographic ranges. However, inference from comparative methods has been limited by uncertainty over whether contemporary biodiversity data retain the historic signal of speciation. We constructed a process-based simulation model to determine the influence of speciation mode and post-speciation range evolution on current biodiversity patterns. The simulations suggest that the signal of speciation history remains detectable in species distributions and phylogeny, even when species ranges have evolved substantially through time. We extracted this signal using a combination of summary statistics that had good power to distinguish speciation modes, then used these statistics to infer the speciation history of 30 plant and animal clades. The results point to broad taxonomic patterns in the modes of speciation, with strongest support for founder speciation in mammals and birds, and strongest support for sympatric speciation in plants. Our model and analyses show that broad-scale comparative methods can be a powerful complementary approach to more focused genomic analyses in the study of the patterns and mechanisms of speciation.

Lineages Through Space and Time in R ([ltstR](https://github.com/alexskeels/ltstR)):

-   During the radiation of a clade, diversification rates can show temporal patterns such as a speedup or slowdown which might relate to different ecological and evolutionary mechanisms. The temporal dynamics of diversification of whole clades are often visualised as a lineagethrough-time (LTT) plot, which traces the number of reconstructed lineages at different time
    points. However, clades do not radiate evenly across space and may show different temporal dynamics in different regions. As such, a biogeographic approach is required to more completely understand temporal diversification dynamics. Here, I present a tool to extract temporal diversity information across different biogeographic regions from the output of
    commonly used ancestral range estimation models implemented in the R package BioGeoBEARS. The lineages through space and time (LTST) plot allows for visualisation of diversification dynamics in different regions, formatted in an accessible way which can be used for further quantitative analysis.

[waRhol](https://github.com/alexskeels/waRhol)

-   waRhol is an R package containing colour palettes inspired by the paintings of Andy Warhol. It can be directly integrated with ggplot and base R plotting.
